src/engine/idTable/IdTable.h

```diff
@@ -474,15 +502,23 @@ class IdTable {
   // `auto newTable = AD_FWD(oldTable).moveOrClone()` which is equivalent to the
   // pattern `auto newX = AD_FWD(oldX)` where the type is copy-constructible
   // (which `IdTable` is not.).
-  auto moveOrClone() const& requires isCloneable { return clone(); }
-  IdTable&& moveOrClone() && requires isCloneable { return std::move(*this); }
+  CPP_template(typename = void)(
+      requires(isCloneable)) auto moveOrClone() const& {
+    return clone();
+  }
+  CPP_template(typename = void)(
+      requires(isCloneable)) IdTable&& moveOrClone() && {
```

Comment:
This could just be rewritten. Probably when trying out you misplaced some of the many `&&`s here or something else went wrong.

Reply:
yeah, I tried this with gcc13 and this was just showing errors from other parts of the code when compiled with backports on

Reply:
here](https://github.com/ad-freiburg/qlever/pull/1746/commits/2dba5720293bd21c9340b60d88a42d969425d260) is gcc13 fix related to CPP_template usage for overloads or specializations, please share if you think is is needed.

Reply:
That fix is definitely needed, I also have it in some open PRs... GCC11 didn't complain about this, but Clang16 (where I treid it) and obviously GCC13 (from what you say) correctly reject that code. It is time that we merge it in.